libs/language-server/src/grammar/expression.langium

```diff
@@ -10,10 +10,13 @@ import './terminal'
 import './transform'
 
 Expression:
-  OrExpression;
+  ReplaceExpression;
 
 // The nesting of the following rules implies the precedence of the operators:
 
+ReplaceExpression infers Expression:
+  OrExpression ({infer TernaryExpression.first=current} operator='replace' second=OrExpression 'with' third=OrExpression)*;
```

Comment:
I am not sure how comfortable I am that I understood this correctly. I think we should try to refactor this grammar to be more clearly readable (e.g. move all binary / ternary expressions into one type, have an easy and clear way to define order of execution instead of this weird setup of referring to the next lower expression).

In addition, I think the order I chose here means that the replace expression has the lowest priority in evaluation possible. Which I think makes sense (evaluate all other expressions first, then use their output for replacing)... but not sure.